app/1.0.x/plugin-development/custom-entities.md

```diff
@@ -129,76 +129,105 @@ To see a real-life example, give a look at the [Key-Auth plugin migrations](http
+    { key = { type = "string", required = false, unique = true, auto = true }, },
+  },
 }
 
-return {keyauth_credentials = SCHEMA} -- this plugin only results in one custom DAO, named `keyauth_credentials`
-```
-
-Since your plugin might have to deal with multiple custom DAOs (in the case
-when you want to store several entities), this module is bound to return a
-key/value table where keys are the name on which the custom DAO will be
-available in the DAO Factory.
-
-You will have noticed a few new properties in the schema definition (compared
-to your [schema.lua]({{page.book.chapters.plugin-configuration}}) file):
-
-| Property name         | Lua type                  | Description
-|-----------------------|---------------------------|-------------
-| `primary_key`         | Integer indexed table     | An array of each part of your column family's primary key. It also supports **composite keys**, even if all Kong entities currently use a simple `id` for usability of the [Admin API]. If your primary key is composite, only include what makes your **partition key**.
-| `fields.*.dao_insert_value` | Boolean              | If true, specifies that this field is to be automatically populated by the DAO (in the base_dao implementation) depending on it's type. A property of type `id` will be a generated uuid, and `timestamp` a timestamp with second-precision.
-| `fields.*.queryable`  | Boolean                   | If true, specifies that Cassandra maintains an index on the specified column. This allows for querying the column family filtered by this column.
-| `fields.*.foreign`    | String                    | Specifies that this column is a foreign key to another entity's column. The format is: `dao_name:column_name`. This makes it up for Cassandra not supporting foreign keys. When the parent row will be deleted, Kong will also delete rows containing the parent's column value.
-
-Your DAO will now be loaded by the DAO Factory and available as one of its
-properties. Since the DAO Factory is exposed by the [Plugin Development Kit]'s
-`kong` global (see [kong.dao](/{{page.kong_version}}/pdk/#kong-dao), you can
-retrieve it like so:
-
-```lua
-local key_credential, err = kong.dao.key_credentials:insert({
-  consumer_id = consumer.id,
-  key = "abcd"
-})
+return { credentials }
 ```
 
-The DAO name (`keyauth_credentials`) with which it is accessible from the DAO
-Factory depends on the key with which you exported your DAO in the returned
-table of `daos.lua`.
-
-You can see an example of this in the [Key-Auth `daos.lua` file](https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/daos.lua).
+The plugin with this `daos.lua` file includes one schema called `keyauth_credentials`.
+
+The DAO for such a Schema would be `kong.db.keyauth_credentials`. It would work like any
+other core DAO: Inserting would be done
+with `kong.db.keyauth_credentials:insert({ consumer = { id = some_consumer_id } })`,
+and so on.
```

Comment:
I agree that the new DAO is probably the most important part of this doc.

Since we currently don't have DAO docs, I have decided to start documenting them on this PR as well.

It does not document all the methods in the DAO, but at least it documents some of them.